presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -416,12 +447,54 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     failureDetector,
                     nodeTaskMap,
                     stageSchedulers,
-                    stageLinkages);
+                    stageLinkages,
+                    stagedDependencies,
+                    schedulingQueue);
             stages.addAll(subTree);
 
             SqlStageExecution childStage = subTree.get(0);
             childStagesBuilder.add(childStage);
         }
+
+        for (Reference<StageExecutionPlan> dependency : plan.getDependencies()) {
```

Comment:
We probably will discuss whether we need a standalone `SuperStage` class. For now I need to distinguish between "inputStage" and "dependentStage": one exchange data via Remote Exchange, another through Disk. It can also be modeled as different stage linkage type...